modern-cpp-font-lock.el

```diff
@@ -53,85 +53,104 @@
   "Provides font-locking as a Minor Mode for Modern C++"
   :group 'faces)
 
-(defun modern-c++-string-lenght< (a b) (< (length a) (length b)))
-(defun modern-c++-string-lenght> (a b) (not (modern-c++-string-lenght< a b)))
-
-(defcustom modern-c++-types
-  (eval-when-compile (sort '("bool" "char" "char16_t" "char32_t" "double" "float" "int" "long" "short" "signed" "unsigned" "void" "wchar_t")
-                           'modern-c++-string-lenght>))
+(eval-and-compile
+  (defun modern-c++-string-lenght< (a b) (< (length a) (length b)))
+  (defun modern-c++-string-lenght> (a b) (not (modern-c++-string-lenght< a b))))
+
```

Comment:
Indeed compiled time variable is better. However, why do you create an indirection with all `--` variables? The compiled time variable should be in a `defvar` scope instead of a `defcustom`?

The notation `--` is not readable friendly. Is it a [convention](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html)?